backend/src/shared/abstract.service.ts

```diff
@@ -1,27 +1,35 @@
-import { Injectable } from '@nestjs/common';
-import { Repository } from 'typeorm';
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { DeleteResult, Repository, UpdateResult } from 'typeorm';
 
 @Injectable()
-export abstract class AbstractService {
-	protected constructor(protected readonly repository: Repository<any>) {}
+export abstract class AbstractService<T> {
+	protected constructor(protected readonly repository: Repository<T>) {}
 
-	async all(): Promise<any[]> {
+	async all(): Promise<T[]> {
 		return this.repository.find();
 	}
 
-	async create(data): Promise<any> {
+	async create(data): Promise<T> {
 		return this.repository.save(data);
 	}
 
-	async findOne(condition): Promise<any> {
-		return this.repository.findOne(condition);
+	async findOne(condition) {
```

Comment:
it seems like this can still have the return type, or is there a reason it isn't on this method?

```ts
async create(data): Promise<T> {
```